stdlib/lazy.mli

```diff
@@ -43,7 +43,8 @@ type 'a t = 'a CamlinternalLazy.t
    [Lazy.t] instead.
 
    Note: [Lazy.force] is not thread-safe.  If you use this module in
```

Comment:
I'm not sure if I agree with the use of "not thread-safe" here. The definition of thread-safety is vague, but I tend to view it as not having a defined semantics. Whereas here, we do have a defined semantics for concurrent forcing. If two domains concurrently force lazy, the program would not crash (quite important this one) and you'll get `Undefined` exception. Hence, I would not mark this API as not thread safe, but still point to `try_force`.

Reply:
It is true that your work has made the API thread-safe, so I will mark this in the documentation. IIUC the trunk implementation, and assuming the native behaviour for polling, there is a race where the lazy_t value could end up with a forward tag, but the first cell concurrently replaced with a closure raising an exception, breaking type safety. So it seems correct to me that previously, thread-unsafe could mean "could crash".

Reply:
Agree with your assessment reg trunk.